File: zlh/LeetCode/2023-11-19/111. 二叉树的最小深度.ts
function minDepth(root: TreeNode | null): number {
  let deep: number = 0;
  const queue: TreeNode[] = [root];

  if (!root) return deep;
  while (queue.length) {
    let len: number = queue.length;
    deep++;

    while (len--) {
      let cur = queue.shift();
      if (!cur.left && !cur.right) return deep;

      cur.left && queue.push(cur.left);
      cur.right && queue.push(cur.right);
    }
  }

  return deep;
}

File: zjj/111. 二叉树的最小深度.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def minDepth(self, root: TreeNode[TreeNode]) -> int:
        if not root:
            return 0
        # 其中一个节点为空，说明左右孩子有一个必然为0，所以可以返回两者和+1
        if root.left == None or root.right == None:
            return self.minDepth(root.left) + self.minDepth(root.right) + 1
        return min(self.minDepth(root.left), self.minDepth(root.right)) + 1
